ebos/eclproblem.hh

```diff
@@ -2932,80 +2932,44 @@ private:
         return eclState.fieldProps().get_global_int(prop);
     }
 
-    void updatePvtnum_()
+    template<class T>
```

Comment:
Why `template`?  Since this internally uses `has_int_prop()` and `get_int_prop()` the function itself isn't usable for anything other than `T = int`.

Reply:
because pvtnum is std::vector<int> and the others are std::vector<unsigned short>

Reply:
>> Why `template`?
>
> because pvtnum is `std::vector<int>` and the others are `std::vector<unsigned short>`

Okay, but then this probably needs some additional work before it's ready.  In particular since `numData` will unconditionally be `vector<int>`, we should probably make the type conversion explicit when assigning to the result vector (`numbers`).  Finally, it's **possible** that the `*NUM` vectors contain zero values in inactive cells in which case   unconditionally subtracting one will be troublesome for the unsigned representations.

Of course these problems already exist in the functions you're replacing, but I think this would be a good opportunity to correct those oversights.

Reply:
i can add the cast but i don't understand what you intend to do about the subtraction. seeing that these cells are inactive, i don't think we would ever read those values later?

Reply:
> i don't understand what you intend to do about the subtraction.

```C++
numbers[elemIdx] = static_cast<T>(std::max(numData[cartElemIdx], 1) - 1);
```